doc/architecture/index.rst

```diff
@@ -57,6 +57,20 @@ Architecture
 
     Indexing system used to speed up historical queries.
 
+    ---
+
+    :fa:`address-book` :doc:`tls_internals`
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Internal implementation of TLS communication.
+
+    ---
+
+    :fa:`address-book` :doc:`quic_internals`
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Internal implementation of QUIC communication.
```

Comment:
Up for discussion, but I'd be tempted to add an "(in-progress)" disclaimer to this?

We don't currently have a great distinction in the architecture docs between descriptions of what _is_ implemented, and what we _plan_ to implement, but I think as we expand we should make that distinction!